src/localize-router.service.ts

```diff
@@ -43,8 +44,32 @@ export class LocalizeRouterService {
       let rootSnapshot: ActivatedRouteSnapshot = this.router.routerState.snapshot.root;
 
       this.parser.translateRoutes(lang).subscribe(() => {
-        const url = this.traverseRouteSnapshot(rootSnapshot);
-        
+        let url = this.traverseRouteSnapshot(rootSnapshot);
+
+        if (!this.settings.alwaysSetPrefix) {
+          var urlSegments = url.split('/');
+          //If the default language has no prefix make sure to remove and add it when necessary
+          if (this.parser.currentLang === this.parser.defaultLang) {
+            //Remove the language prefix from url when current language is the default language
+            if (url.indexOf(this.parser.currentLang + '/') === -1) {
```

Comment:
Would this like work if url has somewhere as a substring default language. For example `product/specimen/` includes `en/` though we should not remove it. We should IMO just check if it starts with language `/en/...` or `en/...` and in this case just remove the first segment.